service/storage_proxy.cc

```diff
@@ -5527,10 +5527,10 @@ bool storage_proxy::is_worth_merging_for_range_query(
 future<result<query_partition_key_range_concurrent_result>>
 storage_proxy::query_partition_key_range_concurrent(storage_proxy::clock_type::time_point timeout,
         locator::effective_replication_map_ptr erm,
-        std::vector<foreign_ptr<lw_shared_ptr<query::result>>>&& results,
+        std::vector<foreign_ptr<lw_shared_ptr<query::result>>>& results,
```

Comment:
It's not really needed. The caller holds the variable ready.

It's clearer to convert this to a value parameter. To track the lifetime, you don't need to go up to the caller, instead you see it's safe here.

btw after all this we can probably merge the caller and callee.